lesson25/index.js

```diff
@@ -0,0 +1,136 @@
+const modalInner = document.getElementById("js-modal-inner");
+const submitButton = document.getElementById("js-submit-button");
+
+const openModal = () => {
+    document.getElementById("js-modal-area").classList.add("is-active");
+    bodyElement.classList.add("fixed");
+    document.getElementById("js-modal-contents").scrollTop = 0;
+};
+
+const closeModal = () => {
+    document.getElementById("js-modal-area").classList.remove("is-active");
+    bodyElement.classList.remove("fixed");
+};
+
+checkboxLink.addEventListener("click", openModal);
+checkboxLink.addEventListener("keypress", openModal);
+
+modalCloseButton.addEventListener("click", () => {
+    closeModal();
+    checkFormValidityToEnableSubmitButton();
+});
+
+//モーダル以外の部分を押すとモーダルが閉じる
+document.addEventListener("click", (e) => {
+    if(e.target.classList.contains("js-modal")){
+        closeModal();
+        checkFormValidityToEnableSubmitButton();
+    }
+});
+
+//スクロールが一番下に行ったらチェックボックスをcheckedにする
+const observerTarget = document.getElementById("js-last-sentence");
+const observerOptions = {
+    root: modalInner,
+    threshold: 1.0
+};
+const checkbox = document.getElementById("js-checkbox");
+
+const setCheckedAttributeToCheckbox = ([entry]) => {
+    if (entry.isIntersecting) {
+        checkbox.checked = true;
+        checkbox.disabled = false;
+    }
+};
+
+const observer = new IntersectionObserver(setCheckedAttributeToCheckbox, observerOptions);
+observer.observe(observerTarget);
+
+
+const nameOfInput = document.querySelector(".js-form-name");
+const emailOfInput = document.querySelector(".js-form-email");
+const passwordOfInput = document.querySelector(".js-form-password");
+const formElements = [nameOfInput, emailOfInput, passwordOfInput];
+
+const validationTerms = {
+    name: {
+        upperLimitOfText: 15
+    },
+    password: {
+        pattern: /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z\d]{8,}$/
+    }
+}
+
+const validationOptions = {
+    name: {
+        isValid: () => {
+            return nameOfInput.value.length < validationTerms.name.upperLimitOfText;
+        },
+        errorMessage: "ユーザー名は15文字以内で入力してください",
+    },
+    email: {
+        isValid: () => {
+            return emailOfInput.validity.valid;
+        },
+        errorMessage: "メールアドレスの形式になっていません",
+    },
+    password: {
+        isValid: () => {
+            return validationTerms.password.pattern.test(passwordOfInput.value);
+        },
+        errorMessage: "8文字以上の大小の英数字を交ぜたものにしてください",
+    }
+};
+
+const addInvalidClass = target => target.classList.add("invalid");
+const removeInvalidClass = target => target.classList.remove("invalid");
+const showErrorMessage = target => target.nextElementSibling.textContent = validationOptions[target.id].errorMessage;
+const isNotEmptyOfInput = target => target.value.trim() === ""? false: true;
+const isValidFormInput = target => validationOptions[target.id].isValid()? true: false;
+
```

Comment:
IMO
I don't think it is necessary to specify the return value explicitly. This is because the result of validationOptions[target.id].isValid() is Boolean.
```suggestion
const isValidFormInput = target => validationOptions[target.id].isValid();

```

Reply:
Thank you for your review!
It's true..!
I was not able to do the elementary things.
I fixed it!
[f2e2633](https://github.com/haru-programming/JavaScript-lessons/pull/29/commits/f2e263387edb1e87416cc4b8abb2d8b336eb73cc)